src/rpc/protocol.h

```diff
@@ -29,7 +29,7 @@ enum HTTPStatusCode
 };
 
 //! Bitcoin RPC error codes
-enum RPCErrorCode
+enum class RPCErrorCode
```

Comment:
@luke-jr there would still be a few ```code == static_cast<int>(RPCErrorCode:: ...)```. But I agree it's not that many.

However changing JSONRPCError is kicking the can down to Pair where the conversion to int has to be done, or down to UniValue. It just seems to me like RPCErrorCode implicitly should be treated like an int since it will almost always be converted into one.